### ISSUE TITLE
Unable to Reproduce Results for train_func.py - Loss Becomes NaN

### ISSUE THREAD

**Issue by fmocking** [bozhenhhu/CoupleNet#1]
Hello @bozhenhhu,

I am having trouble reproducing the results for the `train_func.py` script. The training process fails in the first epoch, with the loss value becoming `NaN`. As a result, the model does not train, and the validation and test accuracies remain at their initial values throughout all epochs.

**Training Log**

Here is a sample of the training output, which shows the issue:

```
Epoch: 074, Validation: 0.0008, Test: 0.0035
Epoch: 075, Validation: 0.0008, Test: 0.0035
Epoch: 076, Validation: 0.0008, Test: 0.0035
...
```

**Steps to Reproduce**

1.  Run the `train_func.py` script with the default hyperparameters.
2.  The loss becomes `NaN` during the first epoch.

I have also tried adjusting the batch size to 4, as mentioned in the paper for this task, but the issue persists.

**Request for Clarification**

I noticed a few discrepancies between the hyperparameters described in the paper and those in the `train_func.py` script. Could you please clarify the correct settings?

*   **Optimizer:** The paper mentions using the `Adam` optimizer, but the script uses `SGD`.
*   **Sequential Kernel Size:** The paper specifies a sequential kernel size (`l`) of 11, while the script uses 21.

Any help you could provide to resolve this issue and reproduce the results would be greatly appreciated. Thank you!

**Comment by bozhenhhu** [comment]
Hi, sorry for late reply. I have left school, and did not have resources to validate these experiments again. A loss becoming NaN during training is a common issue with several potential causes. Here are some suggestions to diagnose, 1. Check learning rate: A learning rate that's too high is one of the most common causes. 2. Look at weight initialization. 3. Importantly, examine data preprocessing, ensure your input data is properly formatted. The network codes are primarily based on the great work, [CDConv,](https://github.com/hehefan/Continuous-Discrete-Convolution). This work changes its input, use the complete structure representations.
